src/main/java/org/skriptlang/skript/elements/fireworks/ExprFireworkFlownTime.java

```diff
@@ -0,0 +1,159 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package org.skriptlang.skript.elements.fireworks;
+
+import org.bukkit.entity.Firework;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Timespan;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Firework Flown Time")
+@Description({
+	"How much time that the fireworks have been boosting the player in an <a href='events.html#elytra_boost'>elytra boost event</a>.",
+	"Maximum is the same as time until detonation. Paper names them differently than Spigot."
+})
+@Examples({
+	"if the time until detonation is less than 5 seconds:",
+		"\tadd 2 seconds to time until detonation of event-firework"
+})
+@Since("INSERT VERSION")
+@SuppressWarnings("removal")
+public class ExprFireworkFlownTime extends SimplePropertyExpression<Firework, Timespan> {
+
+	private final static boolean PAPER = Skript.methodExists(Firework.class, "getTicksFlown");
+
+	/**
+	 * Developer note:
+	 * Spigot's methods are {@link Firework#getLife()} and {@link Firework#getMaxLife()} and were added in 1.19.
+	 * Paper has had these methods since 1.18 but were named the following;
+	 * {@link Firework#getTicksFlown()} and {@link Firework#getTicksToDetonate()} respectfully.
+	 * Paper does not agree with Spigot's method naming, so they have Spigot's deprecated.
+	 */
+	static {
+		registerDefault(ExprFireworkFlownTime.class, Timespan.class, "firework ([detonate:max[imum]] life[time]|flown time|detonate:time until detonat(e|ion))", "fireworks");
+	}
```

Comment:
Don't know how I feel about syntax being
```applescript
firework maximum lifetime of %firework%
firework time until detonation of %firework%
firework flown time of %firework% of %firework%
```
additionally `max lifetime` and `time until detonate` don't seem to mean the same thing but are treated the same way